### PR TITLE
🎉 ✨ wizard: fasttrack integration, modular page config

### PR DESCRIPTION
**Pull request by lucasrodes**
## TL; DR
Run

```
etl-wizard fasttrack
```

and check that it works as expected.

### Tasks
- [x] Make apps more modular. Apps are added to the Wizard homepage by means of a config.yml file.
  - https://github.com/owid/etl/issues/2255
- [x] Move Fast-Track to Wizard.
  - https://github.com/owid/etl/issues/2253
- [x] Add metaplay
- [x] Put Wizard pages in a single folder: `apps/wizard/pages/`

### Review guidelines
Several files are affected by this PR, but most are due to file renaming.

- Renaming of `apps/wizard/templating` → `apps/wizard/etl_steps`.
- Fast-track support in Wizard: `apps/wizard/pages/fasttrack/`
  - [ ] `apps/pages/wizard/fasttrack/app.py`: Entrypoint to the app.
  - [ ] `apps/pages/wizard/load.py`: Load the data from a Google sheet or local CSV. (contains former content from `fasttrack.csv` and `fasttrack.sheets` code, and additional.
  - [ ] `apps/pages/wizard/process.py`: Process and submit the new dataset to Grapher. Some of the code could live in `app.py`, but just took some away to reduce length of it.
  - [ ] `apps/pages/wizard/fast_import.py`: Class `Fasttrackimport`, used in `process.py` to submit the new dataset.
  - [ ] `apps/pages/wizard/utils.py`: Utils for the step.
  - [ ] `apps/pages/wizard/fasttrack/cookiecutter/`: Template to create python grapher script.
  - [ ] `apps/pages/wizard/fasttrack/markdown/`: Text shown in the app (mostly guidelines on how to import data)
- Make adding apps/pages to Wizard simpler: `apps/wizard/config/`
  - [ ] `apps/pages/wizard/config/config.yml`: List of Wizard apps. Easily configurable.
  - [ ] `apps/pages/wizard/config/__init__.py`: Load configuration and others.
  - [ ] `apps/pages/wizard/app.py`, `apps/wizard/home.py`, `apps/wizard/cli.py`: Minor changes to make it work with new modular configuration.
- Integrate **Metaplay** into Wizard
  - [ ] `apps/wizard/pages/metaplay.py`: Script with metaplay app. Deprecated and removed the former version from `apps/metadata_playground/` and command `etl-metaplay` (in favour of `etl-wizard metaplay`).

### Future
- Deprecate `walkthrough`.
- Deprecate `fasttrack` (outside of wizard).
- Add guidelines to [our docs](https://docs.owid.io/projects/etl) on how to add a new page/app to Wizard.

## Summary by CodeRabbit

- **New Features**
	- Introduced dynamic table of contents generation in the Wizard app.
	- Added new Fast-track tool functionalities for importing datasets from Google Sheets or local CSV files.
	- Implemented a metadata playground app for enhanced data handling and visualization.
- **Enhancements**
	- Improved logic for page switching in the Wizard app to support various phases and aliases.
	- Enhanced configuration loading and validation for more robust app settings management.
- **Documentation**
	- Updated guidelines and instructions for importing datasets using the Fast-track tool.
	- Removed outdated references to the `metadata_playground` tool, focusing on current functionalities.
- **Refactor**
	- Streamlined app configuration and CLI phase definitions for better maintainability and user experience.

### CODE REVIEW

**Comment by lucasrodes** [comment]
@Marigold The most relevant change is related to `fasttrack`.

If you could just try it as you usually have and see how it feels. Some things are slightly different (e.g. the file difference) since the Streamlit framework has different capabilities.

File: pyproject.toml [hunk 1]
```diff
@@ -149,7 +149,8 @@ exclude = [
     "lib/",
     "etl/steps/archive",
     "etl/snapshots/archive",
-    "apps/wizard/templating/cookiecutter",
+    "apps/wizard/etl_steps/cookiecutter/",
```
Comment (Marigold) [inline pyproject.toml:152]: Let's remove pywebio from pyproject.py.

Comment (lucasrodes) [reply pyproject.toml:152]: what do you think about leaving the old fast-track here for a couple of weeks to see if anything goes wrong with the new version?

**Comment by Marigold** [comment]
@lucasrodes I tested it locally and ran into an error. Instructions:
1. `etl-wizard fasttrack` locally
2. `Existing sheet`, select `DRAFT OWID Donations / latest`
3. Submit

then I ran into an error
```
File "/Users/mojmir/projects/etl/etl/files.py", line 146, in _str_presenter
    if (len(lines) > 1) or (max([len(line) for line in lines]) > 120):
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I think I should be able to fix it (and get familiar with the code as a bonus), just let me know if you're ok with it.

**Comment by lucasrodes** [comment]
hi @Marigold , I think I was getting this error too when using the former fasttrack version with certain google sheets. So I assumed this was not a new issue.

Looks like maybe there is something off with the google sheet? or maybe there is something not accounted for in the code?

Note that you may have to fix this in two places: apps/fasttrack and apps/wizard/pages/fasttrack

**Pull request merged**